src/cluster_asm.c

```diff
@@ -365,6 +369,161 @@ void asmTaskFree(asmTask *task) {
+
+    /* Add destination node ID (40 chars) */
+    serialized = sdscatlen(serialized, task->dest, CLUSTER_NAMELEN);
+    serialized = sdscat(serialized, ":");
+
+    /* Add operation type */
+    serialized = sdscatprintf(serialized, "%s:", task->operation == ASM_IMPORT ?
+                                                 "import" : "migrate");
+
+    /* Add current state */
+    serialized = sdscatprintf(serialized, "%s:", asmTaskStateToString(task->state));
+
+    /* Add slot ranges sds */
+    sds slot_ranges_str = slotRangeArrayToString(task->slot_ranges);
+    serialized = sdscatprintf(serialized, "%s", slot_ranges_str);
+    sdsfree(slot_ranges_str);
+
+    return serialized;
+}
+
+/* Deserialize ASM task information from a string and create a complete asmTask.
+ * Format: "task_id:source_node:dest_node:operation:state:slot_ranges"
+ * Returns a new asmTask on success, NULL on failure. */
+asmTask *asmTaskDeserialize(sds data) {
+    int count, idx = 0;
+    asmTask *task = NULL;
+    if (!data || sdslen(data) == 0) return NULL;
+
+    sds *parts = sdssplitlen(data, sdslen(data), ":", 1, &count);
+    if (count < 6) goto err;
+
+    /* Parse task ID */
+    if (sdslen(parts[idx]) == 0) goto err;
+    task = asmTaskCreate(parts[idx]);
+    if (!task) goto err;
+    idx++;
+
+    /* Parse source node ID */
+    if (sdslen(parts[idx]) != CLUSTER_NAMELEN) goto err;
+    memcpy(task->source, parts[idx], CLUSTER_NAMELEN);
+    idx++;
+
+    /* Parse destination node ID */
+    if (sdslen(parts[idx]) != CLUSTER_NAMELEN) goto err;
+    memcpy(task->dest, parts[idx], CLUSTER_NAMELEN);
+    idx++;
+
+    /* Parse operation type */
+    if (!strcasecmp(parts[idx], "import")) {
+        task->operation = ASM_IMPORT;
+    } else if (!strcasecmp(parts[idx], "migrate")) {
+        task->operation = ASM_MIGRATE;
+    } else {
+        goto err;
+    }
+    idx++;
+
+    /* Parse state */
+    task->state = ASM_NONE; /* Default state */
+    for (int state = ASM_NONE; state <= ASM_RDBCHANNEL_TRANSFER; state++) {
+        if (!strcasecmp(parts[idx], asmTaskStateToString(state))) {
+            task->state = state;
+            break;
+        }
+    }
+    idx++;
+
+    /* Parse slot ranges */
+    task->slot_ranges = slotRangeArrayFromString(parts[idx]);
+    if (!task->slot_ranges) goto err;
+    idx++;
+
+    sdsfreesplitres(parts, count);
+    return task;
+
+err:
+    if (task) asmTaskFree(task);
+    sdsfreesplitres(parts, count);
+    return NULL;
+}
+
+/* Notify slaves about ASM task information to maintain consistency during slot migration.
+ * This function sends a CLUSTER SYNCSLOTS CONF ASM-TASK command to all connected slaves
+ * with the serialized task information. */
+void asmNotifySlavesStateChange(struct asmTask *task) {
+    if (!server.cluster_enabled || !clusterNodeIsMaster(getMyClusterNode())) return;
+
+    /* Do not propagate migrate task to slaves, as slaves never migrate data. */
+    if (task->operation == ASM_MIGRATE) return;
```

Comment:
Just thinking out loud, currently, we don't propagate migrate operations but the "ASM-TASK" has operation field. This is currently just for future use. So, we can propagate migrate operations if needed without worrying about backward compatibility.

Reply:
if you want and module team want, please let me know

Reply:
No, I don't want and I agree no need to complicate it more. I think it is good what we have now.